src/dsp/sample_analytics.cpp

```diff
@@ -54,19 +71,34 @@ namespace scxt::dsp {
         };
         }
     */
-        float ms = 0.0f;
-        const float divisor = static_cast<float>(s->channels) * static_cast<float>(s->getSampleLength());
-        for(size_t i = 0; i < s->getSampleLength(); i++) {
-            for(int chan = 0; chan < s->channels; chan++) {
-                ms += std::powf(*s->GetSamplePtrF32(chan), 2) / divisor;
+    float ms = 0.0f;
+    const float divisor =
+        static_cast<float>(s->channels) * static_cast<float>(s->getSampleLength());
```

Comment:
You may want to calculate 1/divisor here and multiply by it. Most of our compilers should be able to detect it is unchanged in the loop though so not that big a deal.